src/components/shareable-url-modal/shareable-url-modal.js

```diff
@@ -0,0 +1,173 @@
+import React, { useState } from 'react';
+import { connect } from 'react-redux';
+import { toggleShareableUrlModal } from '../../actions';
+
+import Button from '../ui/button';
+import Input from '../ui/input';
+import LoadingIcon from '../icons/loading';
+import Modal from '../ui/modal';
+
+import './shareable-url-modal.css';
+
+const modalMessages = {
+  default:
+    'Please enter your AWS information and a hosted link will be generated.',
+  failure: 'Something went wrong. Please try again later.',
+  loading: 'Shooting your files through space. Sit tight...',
+  success:
+    'The pipeline has successfully been deployed and the visualisation is hosted via the link below.',
+};
+
+const ShareableUrlModal = ({ onToggle, visible }) => {
+  const [deploymentState, setDeploymentState] = useState('default');
+  const [inputValues, setInputValues] = useState({});
+  const [hasNotInteracted, setHasNotInteracted] = useState(true);
+  const [isLoading, setIsLoading] = useState(false);
+  const [responseUrl, setResponseUrl] = useState(null);
+
+  const onChange = (key, value) => {
+    setHasNotInteracted(false);
+    setInputValues(
+      Object.assign({}, inputValues, {
+        [key]: value,
+      })
+    );
+  };
+
+  const handleSubmit = async () => {
+    setDeploymentState('loading');
+    setIsLoading(true);
+
+    try {
+      const request = await fetch('/api/deploy', {
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        method: 'POST',
+        body: JSON.stringify(inputValues),
```

Comment:
@tynandebold  -- you will have to do the conversion from camel case to snake case and send it as input_values to the backend. 

I looked up online, and that's usually the approach bcoz FE has linting issues is we don't use camelCase and BE has same if we don't use snake_case